### PR TITLE
Adding Care Plan

### PR DESCRIPTION
**Pull request by mitch-lindsay**
https://jira.onemedical.com/browse/PHM-479

The implementation the `Plan of Treatment` section in the CCD follows a different paradigm than the `CarePlan` resource in FHIR.  The CCD has a root that gives the high level text of the plan but the entries are individual actions.  The actions hold all of the details but don't correlate well with the plan.  The FHIR has a single root which contains all of the plan details and references to the associated actions.  I believe the proper way to translate a `Plan of Treatment` would be to create resources of the appropriate type for each entry under the CCD section and they reference them as `actions` associated to the CarePlan.

Thankfully that's not what we have to do to pass the g10 requirements.  g10 requires only a few top level details are filled out in the plan.  Per this documentation (https://fhir.cerner.com/millennium/r4/clinical/care-provision/care-plan/), many of the top level details are even hard-coded because those are the only supported values at this time.  So our work is actually pretty minimal.
For the implementation, I broke the pattern up a bit and didn't use a `Resource` hbs template because everything was getting pulled for the root section details.  
Screen Shot 2022-09-08 at 11 38 06 AM

### CODE REVIEW

File: src/templates/cda/ConsultationNote.hbs [hunk 1]
```diff
@@ -20,6 +20,8 @@
         {{>Sections/Medication.hbs}},
         
         {{>Sections/Procedures.hbs}},
+
```
Comment (mitch-lindsay) [inline src/templates/cda/ConsultationNote.hbs:23]: Not sure which template to add this to so I added it to all of them.

Comment (GrrrD) [reply src/templates/cda/ConsultationNote.hbs:23]: you only need to modify ccd.hbs

File: src/templates/cda/Sections/CarePlan.hbs [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  {{#with (getFirstCdaSectionsByTemplateId msg '2.16.840.1.113883.10.20.22.2.10') as |section|}}
+    {{#with (generateUUID (toJsonString section)) as |Id| }}
+      {
+        "fullUrl":"urn:uuid:{{Id}}",
+        "resource":{
+            "resourceType": "CarePlan",
+            "id":"{{Id}}",
+            "text": {
+              "status": "additional", {{!-- Specifying this additional because most of the structured data below is missing  --}}
+              "div": "<div xmlns=\"http://www.w3.org/1999/xhtml\">{{section.2_16_840_1_113883_10_20_22_2_10.title._}}</div>"
```
Comment (GrrrD) [inline src/templates/cda/Sections/CarePlan.hbs:11]: I think you want to be pulling from the text fields, not the title, i.e. the items under section.2_16_840_1_113883_10_20_22_2_10.text.list.item

Comment (mitch-lindsay) [reply src/templates/cda/Sections/CarePlan.hbs:11]: The challenge is the structure of the `text` field is ad hoc.  In our case it's a list but that's not consistent with other examples I've looked at.  Sometimes it's a HTML table or a div with content.  The ideal solution would be to extract the HTML from the `text` node but I couldn't figure out how to do within handlebars.

Comment (GrrrD) [reply src/templates/cda/Sections/CarePlan.hbs:11]: I think it's fine to assume we'll only ever run this against chirp generated CCDAs and so tailor it to that. I wrote a helper in this pull (not merged yet) https://github.com/onemedical/FHIR-Converter/pull/20/files to extract the text by reference, though in this case I would imagine the thing to do is just grab all the text in that section and shove it in here.

**Review by GrrrD — APPROVED** [review]

**Review by GrrrD — APPROVED** [review]

**Pull request merged**